File: AndroidRest/src/main/java/com/android/rest/domain/Hospital.java
package com.android.rest.domain;

import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.List;


@Data
@Entity
@Table(name="hospital", schema ="afia")
public class Hospital {
	
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Column(unique = true)
	private Long id;
	
	@NotNull
	@Size(min=5, message="Name must be at least 5 characters long")
	@Column(unique = true)
	private String hname;

	private String image;
	
	private String owendby;
	
	private String phoneNumbe;
	
	private String relativeAdress;
	
	private Long latitude;

	private Long longtuide;


	@OneToOne
	@JoinColumn(name ="fx_user")
	private User user;

}

File: AfiyaHospital/app/src/main/java/com/example/afiyahospital/data/domain/Hospital.kt
package com.example.afiyahospital.data

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable

@Entity(tableName = "hospital")
data class Hospital(
    @PrimaryKey(autoGenerate = true) val id:Long,
    val hname:String,
    val image:String,
    val owendby:String,
    val phoneNumbe:String,
    val relativeAdress:String,
    val latitude:String,
    val longtuide : String,
    val user:String
): Serializable {
}